tests/unit/OmiseTest.php

```diff
@@ -1,37 +1,121 @@
+            )
+            ->getMock();
+
+        $this->smarty = $this->getMockBuilder(stdClass::class)
+            ->setMockClassName('Smarty')
+            ->setMethods(
+                array(
+                    'assign',
+                )
+            )
             ->getMock();
 
         $this->omise = new Omise();
+        $this->omise->setSetting($this->setting);
+        $this->omise->setSmarty($this->smarty);
+    }
+
+    public function testConstructor_whenNewTheInstance_theDefaultValueOfTheAttributeSettingMustBeAvailable()
+    {
+        $omise = new Omise();
+
+        $setting = $omise->getSetting();
+
+        $this->assertNotEmpty($setting);
     }
 
-    public function testName_omise()
+    public function testName_theNameThatUsedToReferenceInTheProgramMustBe_omise()
     {
         $this->assertEquals('omise', $this->omise->name);
     }
 
-    public function testDisplayName_Omise()
+    public function testDisplayName_theNameThatUsedToDisplayeToTheMerchantMustBe_Omise()
     {
         $this->assertEquals('Omise', $this->omise->displayName);
     }
 
-    public function testNeedInstance_0()
+    public function testNeedInstance_noNeedToLoadModuleAtTheBackendModulePage()
     {
         $this->assertEquals(0, $this->omise->need_instance);
     }
 
-    public function testBootstrap_true()
+    public function testBootstrap_bootstrapTemplateIsRequired()
     {
         $this->assertEquals(true, $this->omise->bootstrap);
     }
+
+    public function testGetContent_merchantOpenTheSettingPage_retrieveSettingDataFromTheDatabaseAndDisplayOnThePage()
+    {
+        $this->setting->method('getLivePublicKey')->willReturn('live_public_key');
+        $this->setting->method('getLiveSecretKey')->willReturn('live_secret_key');
+        $this->setting->method('isModuleEnabled')->willReturn('module_status');
+        $this->setting->method('isSandboxEnabled')->willReturn('sandbox_status');
+        $this->setting->method('getSubmitAction')->willReturn('submit_action');
+        $this->setting->method('getTestPublicKey')->willReturn('test_public_key');
+        $this->setting->method('getTestSecretKey')->willReturn('test_secret_key');
+        $this->setting->method('getTitle')->willReturn('title');
+
+        $this->smarty->expects($this->exactly(8))
+            ->method('assign')
+            ->withConsecutive(
+                array('live_public_key', 'live_public_key'),
+                array('live_secret_key', 'live_secret_key'),
+                array('module_status', 'module_status'),
+                array('sandbox_status', 'sandbox_status'),
+                array('submit_action', 'submit_action'),
+                array('test_public_key', 'test_public_key'),
+                array('test_secret_key', 'test_secret_key'),
+                array('title', 'title')
+            );
+
+        $this->omise->getContent();
+    }
+
+    public function testGetContent_merchantSaveSetting_theSettingDataMustBeSaved()
```

Comment:
@guzzilar The current expectation part, `theSettingDataMustBeSaved`, is the abstraction level. It has been abstracted from the implementation.

From an example, `theSettingSaveMethodMustBeTriggered`, if the implementation has been changed from `$setting->save()` to `$configuration->update()`, the expectation part need to be changed to conform with the implementation. That is double change (the implementation and the test case name).

To make more example, with the pattern from an example applied to the condition or situation part of the test case name, the current situation part is `merchantSaveSetting`. It might be changed to `submitIsTrue` or others. From that change, it is not guide to any part of how program, process or step work.

But with the abstraction level, the implementation can be changed to anything. The expectation of the unit is still the same, `theSettingDataMustBeSaved`.

But I did not say everything or every test case name can be or must be abstracted. It still need to be considered.

1. How about the beyond benefit of the unit test case name can capture, guide or imply the business logic or requirement objective?

2. Can you provide another example of the expectation part?